clang/test/SemaSYCL/intel-fpga-local.cpp

```diff
@@ -554,6 +587,17 @@ void diagnostics()
   //expected-note@+1 {{did you mean to use 'intel::bank_bits' instead?}}
   [[intelfpga::bank_bits(2, 3, 4, 5)]] unsigned int bankbits[64];
 
+  //CHECK: VarDecl{{.*}} bb_reg 'unsigned int [4]'
+  //CHECK: IntelFPGANumBanksAttr{{.*}}Implicit
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 4
+  //CHECK: IntelFPGAMemoryAttr{{.*}}Implicit memory Default
+  //CHECK: IntelFPGABankBitsAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 2
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 2
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 3
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 3
   //expected-error@+2 1{{'fpga_register' and 'bank_bits' attributes are not compatible}}
   [[intel::bank_bits(2, 3)]]
   [[intel::fpga_register]]
```

Comment:
@AaronBallman, thanks for the reviews. There is an existing test where we are not checking implicit Numbank Attribute during incompatible check with register and bank_bits attribute. I have added AST part to show that bank bits attribute generates  implicit Numbank but does not get checked. Is this sufficient for the test coverage?